_posts/2015-11-25-Property-based-testing-FsCheck-Resharper-NCrunch-Visual-Studio-2013.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: post
+name: property-based-testing-with-FsCheck-Resharper-and-NCrunch-in-Visual-Studio-2013
+title: 'Property based testing with FsCheck, Resharper and NCrunch in Visual Studio 2013'
+date: 2015-11-25 09:00:00 +00:00
+author: Alessandro Di Gioia
+image:
+    src: "/assets/img/custom/blog/2015-12-01-Faran-crane-truck.jpg"
+    attribution:
+        text: Faran crane truck (13) by Aleksander Stein
+        href: https://www.flickr.com/photos/aleksander_se/4395710462/in/photolist-7Grazq-u8FeRe-aqN7wn-quFTva-59p1jo-fCcGpb-7pZsgL-wQaXfD-8dcvpw-7GravJ-55WvHY-mQNY6-cavBq7-tatNgX-ybwDP-73JJXP-zbo2F9-x74s1Q-5RuhNR-7K2Ppz-59KLFw-6RoBFb-hJCjcV-qMhRKv-oD2dvE-b4TNGV-56ah5u-6bGXrN-e9qotr-4Bz52Z-7K6KNU-ma811D-bBeGjZ-rbmUEf-4TsmqM-ejPJ3o-3YCK6v-55WzgW-iKD8Yb-dCt5Er-55WA8A-cavy1G-8dGnzC-55SmLZ-4bSdwh-9WUchN-4Bz4PF-6UVh7x-cavM6y-6h2Q2n/
+tags:
+- craftsmanship
+- functional programming
+- dotnet
+- fsharp
+- fscheck
+- property based testing
+- visual studio
+---
+
+### Background
+Craftsmanship in software strives for high, predictable, repeatable and affordable quality; this is the reason why we care about design, code reviews and of course testing. 
+Test Driven Development helps us create a flexible implementation that meets the expectations of our clients in terms of features, performance and quality; as professionals we have to master as many tools as possible to reach our goal of steadily providing value. 
+	
+### Property based testing
+The above is the reason why I am exploring Property Based Testing; I like to think about it as *"Let the machine do the heavy lifting for you"* or as [John Huges] says *"Don't write tests, generate them"*.
```

Comment:
Incorrect use of semi colons throughout (!). Here is a guide on when to use. https://writing.wisc.edu/Handbook/Semicolons.html